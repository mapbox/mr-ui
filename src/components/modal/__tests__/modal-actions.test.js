import React from 'react';
import { shallow } from 'enzyme';
import ModalActions from '../modal-actions';

describe('primary only', () => {
  let wrapper;
  let primaryCallback;

  beforeEach(() => {
    primaryCallback = jest.fn();
    wrapper = shallow(
      <ModalActions
        primaryAction={{ text: 'Okay', callback: primaryCallback }}
      />
    );
  });

  test('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('fires primary callback', () => {
    wrapper.find({ 'data-test': 'primary-modal-action' }).prop('onClick')();
    expect(primaryCallback).toHaveBeenCalledTimes(1);
  });
});

describe('primary and secondary', () => {
  let wrapper;
  let primaryCallback;
  let secondaryCallback;

  beforeEach(() => {
    primaryCallback = jest.fn();
    secondaryCallback = jest.fn();
    wrapper = shallow(
      <ModalActions
        primaryAction={{ text: 'Okay', callback: primaryCallback }}
        secondaryAction={{ text: 'Cancel', callback: secondaryCallback }}
      />
    );
  });

  test('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('fires primary callback', () => {
    wrapper.find({ 'data-test': 'primary-modal-action' }).prop('onClick')();
    expect(primaryCallback).toHaveBeenCalledTimes(1);
  });

  test('fires secondary callback', () => {
    wrapper.find({ 'data-test': 'secondary-modal-action' }).prop('onClick')();
    expect(secondaryCallback).toHaveBeenCalledTimes(1);
  });
});

describe('primary, secondary, tertiary', () => {
  let wrapper;
  let primaryCallback;
  let secondaryCallback;
  let tertiaryCallback;

  beforeEach(() => {
    primaryCallback = jest.fn();
    secondaryCallback = jest.fn();
    tertiaryCallback = jest.fn();
    wrapper = shallow(
      <ModalActions
        primaryAction={{ text: 'Okay', callback: primaryCallback }}
        secondaryAction={{ text: 'Cancel', callback: secondaryCallback }}
        tertiaryAction={{ text: 'Give up', callback: tertiaryCallback }}
      />
    );
  });

  test('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('fires primary callback', () => {
    wrapper.find({ 'data-test': 'primary-modal-action' }).prop('onClick')();
    expect(primaryCallback).toHaveBeenCalledTimes(1);
  });

  test('fires secondary callback', () => {
    wrapper.find({ 'data-test': 'secondary-modal-action' }).prop('onClick')();
    expect(secondaryCallback).toHaveBeenCalledTimes(1);
  });

  test('fires tertiary callback', () => {
    wrapper.find({ 'data-test': 'tertiary-modal-action' }).prop('onClick')();
    expect(tertiaryCallback).toHaveBeenCalledTimes(1);
  });
});

describe('primary and tertiary', () => {
  let wrapper;
  let primaryCallback;
  let tertiaryCallback;

  beforeEach(() => {
    primaryCallback = jest.fn();
    tertiaryCallback = jest.fn();
    wrapper = shallow(
      <ModalActions
        primaryAction={{ text: 'Okay', callback: primaryCallback }}
        tertiaryAction={{ text: 'Cancel', callback: tertiaryCallback }}
      />
    );
  });

  test('does not render tertiary button, since secondary is not provided', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
