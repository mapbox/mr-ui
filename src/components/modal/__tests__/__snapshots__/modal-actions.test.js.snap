// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`primary and secondary renders as expected 1`] = `
<div
  className="flex-parent flex-parent--center-main flex-parent--center-cross flex-parent--row-reverse flex-parent--wrap"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Button
      block={false}
      data-test="primary-modal-action"
      disabled={false}
      onClick={[MockFunction]}
      passthroughProps={
        Object {
          "aria-label": "Okay",
        }
      }
      size="medium"
      variant="primary"
    >
      Okay
    </Button>
  </div>
  <div
    className="flex-child flex-child--no-shrink mr12"
  >
    <Button
      block={false}
      data-test="secondary-modal-action"
      disabled={false}
      onClick={[MockFunction]}
      passthroughProps={
        Object {
          "aria-label": "Cancel",
        }
      }
      size="medium"
      variant="discouraging"
    >
      Cancel
    </Button>
  </div>
</div>
`;

exports[`primary and tertiary does not render tertiary button, since secondary is not provided 1`] = `
<div
  className="flex-parent flex-parent--center-main flex-parent--center-cross flex-parent--row-reverse flex-parent--wrap"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Button
      block={false}
      data-test="primary-modal-action"
      disabled={false}
      onClick={[MockFunction]}
      passthroughProps={
        Object {
          "aria-label": "Okay",
        }
      }
      size="medium"
      variant="primary"
    >
      Okay
    </Button>
  </div>
</div>
`;

exports[`primary only renders as expected 1`] = `
<div
  className="flex-parent flex-parent--center-main flex-parent--center-cross flex-parent--row-reverse flex-parent--wrap"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Button
      block={false}
      data-test="primary-modal-action"
      disabled={false}
      onClick={[MockFunction]}
      passthroughProps={
        Object {
          "aria-label": "Okay",
        }
      }
      size="medium"
      variant="primary"
    >
      Okay
    </Button>
  </div>
</div>
`;

exports[`primary, secondary, tertiary renders as expected 1`] = `
<div
  className="flex-parent flex-parent--center-main flex-parent--center-cross flex-parent--row-reverse flex-parent--wrap"
>
  <div
    className="flex-child flex-child--no-shrink"
  >
    <Button
      block={false}
      data-test="primary-modal-action"
      disabled={false}
      onClick={[MockFunction]}
      passthroughProps={
        Object {
          "aria-label": "Okay",
        }
      }
      size="medium"
      variant="primary"
    >
      Okay
    </Button>
  </div>
  <div
    className="flex-child flex-child--no-shrink mr12"
  >
    <Button
      block={false}
      data-test="secondary-modal-action"
      disabled={false}
      onClick={[MockFunction]}
      passthroughProps={
        Object {
          "aria-label": "Cancel",
        }
      }
      size="medium"
      variant="discouraging"
    >
      Cancel
    </Button>
  </div>
  <div
    className="flex-child flex-child--grow flex-child--no-shrink mr12"
  >
    <button
      aria-label="Give up"
      className="link link--gray txt-bold txt-s"
      data-test="tertiary-modal-action"
      onClick={[MockFunction]}
      type="button"
    >
      Give up
    </button>
  </div>
</div>
`;
